### ISSUE TITLE
Warn users when editing a tag that has a source

### ISSUE THREAD

**Issue by Discostu36** [openstreetmap/iD#9453]
### Description

The validator should warn users when they edited the value of a tag that has a source tag associated with it. 

Examples:

````
highway=residential
maxspeed=30
source:maxpeed=DE:zone30
````
The user edits `maxspeed `to `50`.
````
place=city
population=50123
source:popolation=INSEE 2022
````
The user changes `population `to `50342`. 

````
highway=residential
width=6
source:width=ARCore
````
The user changes `width `to `5.5`

In each of those cases, a warning message should be shown by the validator, because it is very likely, that the source has changed. In some cases like the first example, it might also call attention to a tagging mistake. A first draft of how such a message could be, see below.

### Screenshots

![grafik](https://user-images.githubusercontent.com/30999326/211171414-ad7fabff-ae7f-4c06-81d9-b8fc9d277d82.png)

**Comment by tordans** [comment]
I agree, that this is an issue in areas that are highly mapped. I have fixed miss-matches regularly when new users change the `maxspeed` but – understandably – don't see or understand the source* fields under "tags".

---

On the topic of `maxspeed`, there is the issue of the badly specified source fields. iD would either have to only support one field… or all… or wait for a cleanup of the tagging schema. https://github.com/westnordost/osm-legal-default-speeds/issues/4 shows a few of those tags and the linked video (last comment) goes into more details.

---

A good fist step to make those fields more visible would IMO be to add them as optional `moreTags` field via https://github.com/openstreetmap/id-tagging-schema. This would show them when present, but only then. `source:population` already exists ins this way https://github.com/openstreetmap/id-tagging-schema/blob/main/data/fields/source/population.json. In this case it is visible as soon as population is selected (see `prerequisiteTag`). Which IMO makes sense here but I am unsure about the `maxspeed` source field; we would have to make sure the inputs are well defined for new users to pick from.

---

Broader picture: I see a pattern here of "supporting tags" that would ideally be displayed along side the main tag. Maybe some time iD can have a UI that shows them together. Right now I see
- `source:maxspeed` tag
- `check_date:maxspeed` tag – see references in https://github.com/openstreetmap/iD/issues?q=is%3Aissue+is%3Aopen+check_date
- `description:maxspeed`, `fixme:maxspeed` or `note:maxspeed` tags (text tags that are referenced to one specific tag)

**Comment by tyrasd** [comment]
This would definitely be quite useful, but maybe less as a validation function but rather like @tordans outlines in form of proper support in the fields' UI elements.

The tricky part is going to create the user interface in a way that it remains intuitive, easy to use and not overloaded with meta-information (thus potentially distracting from the important information in the main tag). It's also a good idea to also think about `check_date` and other related tags at the same time.

**Comment by 1ec5** [comment]
As I mentioned in https://github.com/openstreetmap/id-tagging-schema/issues/1469#issuecomment-2694949309, OHM’s iD fork treats sources as a built-in aspect of the tagging scheme. OHM doesn’t currently warn if a source is missing or if a value changes without changing the source. However, you can tag a `*:source=*` (yes, backwards from OSM) by clicking the “@” icon at the top of any field. The source is listed immediately below the value, making it less likely to get out of sync with the value.

<img width="400" height="152" alt="Name is Chicken Steps, and Source for Name is a URL." src="https://github.com/user-attachments/assets/68b40879-721b-4abc-ad10-ee0ac87b2620" />

OHM didn’t upstream this feature because it was unclear if the OSM community wanted sources to be so prominent in the UI that users might feel compelled to tag them routinely. But if this feature is considered a good tradeoff for ensuring internally consistent data, then it should be pretty straightforward to upstream.
